config/src/main/java/com/yahoo/config/subscription/impl/JRTManagedConnectionPools.java

```diff
@@ -0,0 +1,56 @@
+package com.yahoo.config.subscription.impl;
+
+import com.yahoo.config.subscription.ConfigSourceSet;
+import com.yahoo.vespa.config.JRTConnectionPool;
+import com.yahoo.vespa.config.TimingValues;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+public class JRTManagedConnectionPools {
+    private static class JRTSourceThreadFactory implements ThreadFactory {
+        @Override
+        public Thread newThread(Runnable runnable) {
+            Thread t = new Thread(runnable, String.format("jrt-config-requester-%d", System.currentTimeMillis()));
+            // We want a daemon thread to avoid hanging threads in case something goes wrong in the config system
+            t.setDaemon(true);
+            return t;
+        }
+    }
+    private static class CountedPool {
+        long count;
+        final JRTConnectionPool pool;
+        final ScheduledThreadPoolExecutor scheduler = new ScheduledThreadPoolExecutor(1, new JRTSourceThreadFactory());
+        CountedPool(JRTConnectionPool requester) {
+            this.pool = requester;
+            count = 1;
+        }
+    }
+    private Map<ConfigSourceSet, CountedPool> pools = new HashMap<>();
+
+    public synchronized JRTConfigRequester acquire(ConfigSourceSet sourceSet, TimingValues timingValues) {
+        CountedPool countedPool = pools.get(sourceSet);
+        if (countedPool == null) {
+            countedPool = new CountedPool(new JRTConnectionPool(sourceSet));
+            pools.put(sourceSet, countedPool);
+        }
+        countedPool.count++;
```

Comment:
Won't count be 2 then for a newly created CountedPool? So ContedPool should either start count on 0 or only increment in this place when countedPool is not null?

Reply:
Good catch. Test added and fixed.